src/parser/parseUseDirectives.ts

```diff
@@ -1,79 +1,81 @@
 
-    var xlink = xlinkAttribute.slice(1),
-      x = el.getAttribute('x') || 0,
-      y = el.getAttribute('y') || 0,
-      el2 = elementById(doc, xlink).cloneNode(true),
+    const xlink = xlinkAttribute.slice(1),
+      x = el.getAttribute('x') ?? '0',
+      y = el.getAttribute('y') ?? '0',
+      namespace = svgNS;
+    let el2 = elementById(doc, xlink)?.cloneNode(true) as Element,
       currentTrans =
-        (el2.getAttribute('transform') || '') +
+        (el2?.getAttribute('transform') ?? '') +
         ' translate(' +
         x +
         ', ' +
         y +
         ')',
-      parentNode,
-      oldLength = nodelist.length,
-      attr,
-      j,
-      attrs,
-      len,
-      namespace = svgNS;
+      parentNode: ParentNode;
 
     applyViewboxTransform(el2);
     if (/^svg$/i.test(el2.nodeName)) {
       const el3 = el2.ownerDocument.createElementNS(namespace, 'g');
-      for (j = 0, attrs = el2.attributes, len = attrs.length; j < len; j++) {
-        attr = attrs.item(j);
-        el3.setAttributeNS(namespace, attr.nodeName, attr.nodeValue);
-      }
+      Array<Attr>
+        .from(el2.attributes)
+        .forEach((attr: Attr) =>
+          el3.setAttributeNS(namespace, attr.nodeName, attr.nodeValue ?? '')
+        );
       // el2.firstChild != null
       while (el2.firstChild) {
         el3.appendChild(el2.firstChild);
       }
       el2 = el3;
     }
 
-    for (j = 0, attrs = el.attributes, len = attrs.length; j < len; j++) {
-      attr = attrs.item(j);
+    for (let j = 0, attrs = el.attributes, len = attrs.length; j < len; j++) {
+      const attr = attrs.item(j);
       if (
-        attr.nodeName === 'x' ||
-        attr.nodeName === 'y' ||
-        attr.nodeName === 'xlink:href' ||
-        attr.nodeName === 'href'
+        attr?.nodeName === 'x' ||
+        attr?.nodeName === 'y' ||
+        attr?.nodeName === 'xlink:href' ||
+        attr?.nodeName === 'href'
       ) {
         continue;
       }
 
-      if (attr.nodeName === 'transform') {
+      if (attr?.nodeName === 'transform') {
         currentTrans = attr.nodeValue + ' ' + currentTrans;
-      } else {
-        el2.setAttribute(attr.nodeName, attr.nodeValue);
+      } else if (attr) {
+        el2.setAttribute(attr.nodeName, attr.nodeValue ?? '');
       }
     }
 
     el2.setAttribute('transform', currentTrans);
     el2.setAttribute('instantiated_by_use', '1');
     el2.removeAttribute('id');
-    parentNode = el.parentNode;
-    parentNode.replaceChild(el2, el);
-    // some browsers do not shorten nodelist after replaceChild (IE8)
-    if (nodelist.length === oldLength) {
-      i++;
+    if (el.parentNode) {
+      parentNode = el.parentNode;
+      parentNode.replaceChild(el2, el);
     }
+    // TODO: ask about this
+    // please tell me if deleting/changing this will affect functionality
+    // // some browsers do not shorten nodeList after replaceChild (IE8)
+    // if (nodeList.length === oldLength) {
+    //   nodeIdx++;
+    // }
```

Comment:
we can trash patches for buggy old browsers.
IE8 is not supported since long, i m not sure how JSDOM behaves, but we should have visual tests or tests for this case that would fail under node